File: WebApplication10/Models/Customer.cs
using System;
using System.Collections.Generic;

namespace WebApplication10.Models
{
    public partial class Customer
    {
        public Customer()
        {
            Contacts = new HashSet<Contact>();
            Departments = new HashSet<Department>();
            Users = new HashSet<User>();
        }

        public int CustomerId { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public string Comments { get; set; }
        public string Type { get; set; }
        public int? NumberOfSchools { get; set; }
        public ICollection<Contact> Contacts { get; set; }
        public ICollection<Department> Departments { get; set; }
        public ICollection<User> Users { get; set; }
    }
}

File: WebApplication10/ClientApp/app/components/Models/Customer.ts
﻿import { Contact } from "./contact";
import { Department } from "./department"
import { User } from "./user";

export class Customer {

   
    constructor(
        public CustomerId?: number,
        public name?: string,
        public address?: string,
        public email?: string,
        public phone?: string,
        public comments?: string,
        public contacts: Contact[] = [],
        public departments: Department[] = [],
        public users: User[] = [] ) { }
}